### DISCUSSION TITLE
Room/Floor presence detection with only 1 Raspberry Pi

### DISCUSSION THREAD

**Discussion by frankiej911** [mKeRix/room-assistant#616] [Q&A] [answered]
Hello,

Is it possible to use a single Raspberry Pi (I have one in the living room as an audio streamer) to calculate if a device is nearby, if so I'm in the living room, if not I'm upstairs or somewhere else. With the RSSI values it is possible to find out the signal strength, so if a device is -> -90/100dBm, then it's far away. Home Assisstant still knows that I'm home, beceause my device tracker is based on GPS with the Home Assistant app on my iPhone. Then I can make some if/else automations in Home Assistant looking at the RSSI signal strengths and other values (motion, GPS...).

Thanks

**Comment by mKeRix** [comment, marked as answer]
You can do that with room-assistant and some templating in Home Assistant, yeah. You can set a `maxDistance`/`minRssi` (option name depending on integration) which will do the distance limitation for you, or you can just use the distance attribute of the entity in Home Assistant.

**Reply by frankiej911** [reply]
Thanks! I will take a look. Maybe it's not working as good, so in the meantime I've replaced a RPI Zero W as a Smart water meter sensor with an ESP, so I can use this RPI for Room Assistant upstairs to make a stable and supported set-up :)
